core/src/main/java/org/apache/calcite/sql/fun/SqlCoalesceFunction.java

```diff
@@ -29,12 +29,21 @@
 import org.apache.calcite.sql.validate.SqlValidator;
 import org.apache.calcite.util.Util;
 
+import com.google.common.annotations.VisibleForTesting;
+
 import java.util.List;
 
 /**
  * The <code>COALESCE</code> function.
  */
 public class SqlCoalesceFunction extends SqlFunction {
+
+  /**
+   * Only used for testing.
+   */
+  @VisibleForTesting
+  public static boolean enableRewriteCall = true;
```

Comment:
Is it necessary to disable the rewriting logic? I have impression that there are some test cases that can shot the `CoalesceImplementor`. Maybe you can try to remove it from `RexImpTable` and run tests to find out the ones.

Reply:
I didn't find a better way to give a sql and test the codegen and verify the results;
>  I have impression that there are some test cases that can shot the CoalesceImplementor. Maybe you can try to remove it from RexImpTable and run tests to find out the ones.
I only find some IT in QuidemTest failed, but as I mentioned in the JIRA description, they are generated when optimization. And I think running a COALESCE by sql explicitly would be a better way for testing.

Reply:
Thanks  a lot for review ~ and BTW @DonnyZone , what's your mail address ? It's great if we can some talk sometimes.

Reply:
I have updated the email in my profile.